### PR TITLE
Updated documentation on how to deprecate features

### LINKED ISSUE #1421: Improve documentation on how to deprecate features
In #1416 an optional argument of a preprocessor function was deprecated. There we realized that the [existing documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#backward-compatibility) does not describe the procedure in full detail.

#1416 introduced a [custom](https://github.com/ESMValGroup/ESMValCore/blob/4b8bf66ce6ef51cc6814b5d18357ae91684eb0b7/esmvalcore/exceptions.py#L21-L22) deprecation warning that can be [called](https://github.com/ESMValGroup/ESMValCore/blob/4b8bf66ce6ef51cc6814b5d18357ae91684eb0b7/esmvalcore/preprocessor/_mask.py#L153) using the default python warning framework. The existing documentation needs to be updated to reflect these changes.

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR adds documentation on how to deprecate features. 

Closes #1421

Link to documentation: https://esmvaltool--1426.org.readthedocs.build/projects/ESMValCore/en/1426/contributing.html#backward-compatibility

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: doc/contributing.rst
Comment (zklaus) [inline doc/contributing.rst:560]:
```suggestion
:class:`~esmvalcore.exceptions.ESMValCoreDeprecationWarning` using the
:mod:`warnings` module and schedule it for removal two `minor versions
```
Prefixing the fully qualified class name (FQCN) with the `~` will not show the containing packages, thus making it a bit more friendly to read. `:py:` can be omitted because it is [the default domain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#basic-markup).

**Review by zklaus — APPROVED** [review]
LGTM. I left one small sphinx comment. I would like to have approval from @bouweandela before merging.

File: doc/contributing.rst
Comment (valeriupredoi) [inline doc/contributing.rst:563]:
```suggestion
that feature should be removed in version 2.7:
```

**Review by valeriupredoi — APPROVED** [review]
same as Klaus - one minor change comment, and would be happy if @bouweandela saw this first before merge, cheers for this, Manu :beer:

**Comment by schlunma** [comment]
I will merge this by the end of this week if there are no further objections :+1:

File: doc/contributing.rst
Comment (bouweandela) [inline doc/contributing.rst:557]: Please [always start a new sentence on a new line in rst files](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation). This results in much more readable diffs.

**Review by bouweandela — APPROVED** [review]
Looks fine to me @schlunma, but please see the recommendation on formatting .rst files.

**Pull request merged**
